src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/resolver/CoreScalarResolverTest.kt

```diff
@@ -0,0 +1,14 @@
+package it.krzeminski.snakeyaml.engine.kmp.resolver
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import it.krzeminski.snakeyaml.engine.kmp.nodes.Tag
+
+class CoreScalarResolverTest: FunSpec({
+    test("resolve implicit integer") {
+        CoreScalarResolver.INT.matches("0o1010") shouldBe true
```

Comment:
TODO: add clues to make the failed assertions more readable.

Reply:
Or a custom marcher?
```suggestion
        CoreScalarResolver.INT shouldMatch "0o1010"
```

Reply:
Why custom?
What is wrong with the built-in `io.kotest.matchers.regex.shouldMatch` and `io.kotest.matchers.regex.shouldNotMatch` I used in my version?

Reply:
Oh, I rushed this PR and didn't notice that `matches` deals just with `Regex`, and it's not our method...